controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/DeleteScopeTask.java

```diff
@@ -0,0 +1,165 @@
+import io.pravega.client.stream.Stream;
+import io.pravega.client.stream.impl.StreamImpl;
+import io.pravega.common.Exceptions;
+import io.pravega.common.concurrent.Futures;
+import io.pravega.common.tracing.TagLogger;
+import io.pravega.common.util.AsyncIterator;
+import io.pravega.common.util.ContinuationTokenAsyncIterator;
+import io.pravega.controller.store.kvtable.KVTableMetadataStore;
+import io.pravega.controller.store.stream.OperationContext;
+import io.pravega.controller.store.stream.StreamMetadataStore;
+import io.pravega.controller.task.Stream.StreamMetadataTasks;
+import io.pravega.shared.controller.event.DeleteScopeEvent;
+import org.slf4j.LoggerFactory;
+
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static io.pravega.shared.NameUtils.READER_GROUP_STREAM_PREFIX;
+
+/**
+ * Request handler for executing a delete operation for a ReaderGroup.
+ */
+public class DeleteScopeTask implements ScopeTask<DeleteScopeEvent> {
+    private static final TagLogger log = new TagLogger(LoggerFactory.getLogger(DeleteScopeTask.class));
+
+    private final StreamMetadataStore streamMetadataStore;
+    private final StreamMetadataTasks streamMetadataTasks;
+    private final KVTableMetadataStore kvtMetadataStore;
+    private final ScheduledExecutorService executor;
+
+    public DeleteScopeTask(final StreamMetadataTasks streamMetadataTasks,
+                           final StreamMetadataStore streamMetaStore,
+                           final KVTableMetadataStore kvtMetadataStore,
+                           final ScheduledExecutorService executor) {
+        Preconditions.checkNotNull(streamMetaStore);
+        Preconditions.checkNotNull(executor);
+        this.streamMetadataStore = streamMetaStore;
+        this.streamMetadataTasks = streamMetadataTasks;
+        this.kvtMetadataStore = kvtMetadataStore;
+        this.executor = executor;
+    }
+
+    @Override
+    public CompletableFuture<Void> execute(final DeleteScopeEvent request) {
+        String scope = request.getScope();
+        long requestId = request.getRequestId();
+        final OperationContext context = streamMetadataStore.createScopeContext(scope, requestId);
+        log.debug(requestId, "Deleting {} scope", scope);
+        return streamMetadataStore.checkScopeExists(scope, context, executor).thenCompose( exists -> {
+            if (!exists) {
+                log.warn(requestId, "Scope not found: {}", scope);
+                throw new NoSuchScopeException();
+            }
+            return CompletableFuture.completedFuture(null);
+        }).thenCompose(v1 -> {
+            try {
+                deleteScopeContent(scope, context, requestId);
```

Comment:
Instead of 2 separate methods `deleteScopeContent` and `deleteScopeRecursive` I'd rather invoke only 1 that does everything required to delete the scope recursively.

Reply:
Code refraction required for this has been completed now